compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -170,19 +171,47 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
                         .position(|arg| arg.id() == hir_id)
                         .map(|index| (index, seg))
                 });
+            // FIXME(associated_const_equality): recursively search through the bindings instead
+            // of just top level.
+
             let (arg_index, segment) = match filtered {
                 None => {
-                    tcx.sess
-                        .delay_span_bug(tcx.def_span(def_id), "no arg matching AnonConst in path");
-                    return None;
+                    let binding_filtered = path
+                        .segments
+                        .iter()
+                        .filter_map(|seg| seg.args.map(|args| (args.bindings, seg)))
+                        .find_map(|(bindings, seg)| {
+                            bindings
+                                .iter()
+                                .filter_map(|binding| {
```

Comment:
No need for `filter_map`, you can do the entire logic inside `position` (I know this is pre-existing code you copied, but still ^^ feel free to adjust the pre-existing code, too).

Reply:
I think this would be different logically, since `position` would change after the `filter_map`?

Reply:
oof, right. that is some weird subtle logic in this entire function...